modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/impl/PageMemoryNoStoreImpl.java

```diff
@@ -167,23 +169,26 @@ public class PageMemoryNoStoreImpl implements PageMemory {
     /**
      * Constructor.
      *
-     * @param directMemoryProvider Memory allocator to use.
      * @param dataRegionConfig Data region configuration.
      * @param ioRegistry IO registry.
      * @param pageSize Page size in bytes.
      */
     public PageMemoryNoStoreImpl(
-            DirectMemoryProvider directMemoryProvider,
             PageMemoryDataRegionConfiguration dataRegionConfig,
             PageIoRegistry ioRegistry,
             // TODO: IGNITE-17017 Move to common config
             int pageSize
     ) {
-        this.directMemoryProvider = directMemoryProvider;
         this.ioRegistry = ioRegistry;
         this.trackAcquiredPages = false;
         this.dataRegionConfigView = dataRegionConfig.value();
 
+        if (dataRegionConfigView.memoryAllocator() instanceof UnsafeMemoryAllocator) {
```

Comment:
Did you forgot to invert the condition? Or am I missing something?

Reply:
Yes, I made a mistake, I fixed it